confluent_kafka/src/Producer.c

```diff
@@ -402,20 +402,20 @@ static PyObject *Producer_init_transactions (Handle *self, PyObject *args) {
         CallState cs;
         char errstr[256];
         rd_kafka_resp_err_t err;
-        double tmout = 1;
+        double tmout = -1;
 
         if (!PyArg_ParseTuple(args, "|d", &tmout))
                 return NULL;
 
         CallState_begin(self, &cs);
 
-        err = rd_kafka_init_transactions(self->rk, (int)(tmout * 1000), errstr, sizeof(errstr));
+        err = rd_kafka_init_transactions(self->rk, cfl_toMS(tmout), errstr, sizeof(errstr));
 
         if (!CallState_end(self, &cs))
                 return NULL;
 
         if (err != RD_KAFKA_RESP_ERR_NO_ERROR) {
-                cfl_PyErr_Format(err, "Failed to initialize transactions within %.2f seconds", tmout);
+                cfl_PyErr_Format(err, "Failed to initialize transactions within %d seconds: %s", tmout, errstr);
```

Comment:
I dont think you need to prefix the error strings from the transacitonal API, they are rich as-is.